recurring_ical_events.py

```diff
@@ -213,27 +213,32 @@ def __init__(self, component, keep_recurrence_attributes=False):
         self.make_all_dates_comparable()
 
         self.duration = self.end - self.start
-        self.rule = rule = rruleset(cache=True)
-        _rule = component.get("RRULE", None)
-        if _rule:
-            # We don't support multiple RRULE yet, but we can support cases
-            # where the same RRULE is erroneously repeated
-            if isinstance(_rule, list):
-                if len(_rule) > 0 and all(part == _rule[0] for part in _rule):
-                    _rule = _rule[0]
-                else:
-                    raise ValueError("Don't yet support multiple distinct RRULE properties")
-            self.rrule = self.create_rule_with_start(_rule.to_ical().decode())
-            rule.rrule(self.rrule)
-        else:
-            self.rrule = None
+        self.rule = rruleset(cache=True)
+        _component_rules = component.get("RRULE", None)
+        self.until = None
+        if _component_rules:
+            if not isinstance(_component_rules, list):
+                _component_rules = [_component_rules]
+            else:
+                _dedup_rules=[]
```

Comment:
I will just add a few comments in the process and they might not be necessary for merge. They are partially so I understand the code and you know if I misunderstand something or miss it :)

I think, I would add a comment for the purpose here.
```
# Remove duplicates from rules
```

I wonder if deduplication is necessary but definitely nice.